### PR TITLE
Add generic annotation hook

### PR DESCRIPTION
**Pull request by ryanjduffy**
Adds the global fn, `__RECORD_REPLAY_ANNOTATION_HOOK__`, when recording to allow userland scripts to add their own annotations.

### CODE REVIEW

**Comment by ryanjduffy** [comment]
@jcmorrow , @kannanvijayan - here's a first cut at a generic annotation handler. Going to add some input validation too next but wanted to kick off a build while I'm off picking up the kids.

FYI: @bhackett1024, @jasonLaster , and @jaril

File: devtools/server/actors/replay/annotations/contentScript.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        return false;
+      }
+
+      RecordReplayControl.onAnnotation(
+        "generic",
+        JSON.stringify({
+          source,
+          message
+        })
+      );
```
Comment (bhackett1024) [inline devtools/server/actors/replay/annotations/contentScript.js:23]: Why use a "generic" kind for the annotations added here when we could just pass in the kind to the hook?  That seems to add unnecessary complexity.  If the concern is that random scripts would have access to this, I don't think that's much of a problem as if some page wants to mess up their recording with random annotations for existing kinds that's its own problem, and it's easy to see where a given annotation was added by time traveling to its bookmark.

Comment (ryanjduffy) [reply devtools/server/actors/replay/annotations/contentScript.js:23]: My instinct is always to reserve space for internal use when exposing userland methods so we don't have to worry about breaking user customizations when we want to add features later.

That said, I'm not sure this design best suits that purpose. Maybe reserving the `replay-` prefix for the kind would be sufficient.

Comment (ryanjduffy) [reply devtools/server/actors/replay/annotations/contentScript.js:23]: I'm going to simplify and not worry about reserving space. Seems nice but not necessary.

**Review by kannanvijayan — APPROVED** [review]

**Pull request merged**